lib/protocol/message.go

```diff
@@ -148,6 +156,13 @@ type Device struct {
 	Options         []Option // max:64
 }
 
+type FileDownloadProgressUpdate struct {
```

Comment:
As a side note here, if (when) we do decide to change the wire format for requests, we must keep in mind that we will most likely be using variable blocks sizes soon (so index number may not be unambiguous) and probably different hash algorithms, which may not be guaranteed secure (so hash may not be unique). Hence folder+file+offset+size is the easiest way to specify what we need, and hash is included to verify it's the right thing plus give the opportunity to find another source for the same block *if* the hash algo is secure.